### ISSUE TITLE
submit button remains clickable after first submit.

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#35091]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:** needs reproduction
**Reproducible in production?:** needs reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @tylerkaraszewski 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1706118036323349

## Action Performed:
1. Create an expense report by scanning a receipt
2. Click "submit" button
## Expected Result:
Able to submit and submit button shouldn't reappear
## Actual Result:
Submit button remains clickable
## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/43996225/f09c69e2-434c-4205-9949-1457cfee28fa

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by trjExpensify** [comment]
Asking a few Qs: https://expensify.slack.com/archives/C049HHMV9SM/p1706123412357589?thread_ts=1706118036.323349&cid=C049HHMV9SM

**Comment by luacmartins** [comment]
I think the issue involves auto-submission. I've noticed that when the report has already been auto-submitted and then I try to submit again. @tylerkaraszewski, do you see any system messages when clicking into the report preview?

**Comment by trjExpensify** [comment]
[Discussion](https://expensify.slack.com/archives/C02NZ2HGJAZ/p1706634628078839) for ref.

**Comment by mountiny** [comment]
Yes, taking this! Seems like we do send the updates when this comes from harvesting too, will have to dig more

**Comment by trjExpensify** [comment]
The plot thickens.. Thanks!

**Comment by mountiny** [comment]
focusing on wave8 critical issue

**Comment by mountiny** [comment]
still focusing on ideal nav

**Comment by mountiny** [comment]
Still focusing on the ideal nav over this one, hopefully I can make progress on this one this week

**Comment by mountiny** [comment]
Still focusing on managing issues in wave8 and fire today

**Comment by mountiny** [comment]
working at limited capacity - ill

**Comment by mountiny** [comment]
Still focusing on wave8 Critical task over this one, but will tackle this once I get some spare time

**Comment by mountiny** [comment]
In NewDot the fix will be straightforward I believe, we need to check if the originalMessage has harvesting key set to true and then show the updated copy.

To update the notifications, I need to find where this is queued in the backend during harvesting

**Comment by trjExpensify** [comment]
@mountiny is on this one anyway, Melv.

**Comment by mountiny** [comment]
I will need to start storing the amount and currency in the submitted report action `message` so we can compose it in web too and capture the correct price at the time of submission

**Comment by greg-schroeder** [comment]
Moving this one to a `Hot Pick` as discussed last weeek!

**Comment by trjExpensify** [comment]
@mountiny have you dropped this one back into the pool or something?

**Comment by trjExpensify** [comment]
In fact, @JmillsExpensify I think this should go in #wave-control now -- given that the focus of #wave-collect is not having any delayed submission. What do you think?

**Comment by trjExpensify** [comment]
Same as above, Melv.

**Comment by JmillsExpensify** [comment]
Yeah, great point. Filing in #wave-control.

**Comment by trjExpensify** [comment]
Cool, thanks!

**Comment by trjExpensify** [comment]
Still need a volunteer for this one. @JmillsExpensify can you give it a shoutout in the #wave-control update?

**Comment by trjExpensify** [comment]
Hot pick for #wave-control, needs a volunteer still.

**Comment by trjExpensify** [comment]
Put this in your Hot Picks section of the project, @JmillsExpensify.

**Comment by trjExpensify** [comment]
@JmillsExpensify can you call this hot pick out this week?

**Comment by trjExpensify** [comment]
Dropping to weekly when hot pickin'

**Comment by trjExpensify** [comment]
@JmillsExpensify are you calling out hot picks for #wave-control?

**Comment by trjExpensify** [comment]
Hopefully we can get this picked up in wave-control once we're through earnings.

**Comment by trjExpensify** [comment]
Still need a volunteer in #wave-control.

**Comment by JmillsExpensify** [comment]
Haven't been highlighting hot picks in wave-control though I can start doing that this coming week.

**Comment by trjExpensify** [comment]
Love it, thanks!

**Comment by JmillsExpensify** [comment]
Highlighting in the weekly update, as still no takers.

**Comment by trjExpensify** [comment]
Cool, got it.

**Comment by trjExpensify** [comment]
Any bites?

**Comment by JmillsExpensify** [comment]
Still no bites. Our hot picks tied to CRITICAL issues are all the got picked up last week.

**Comment by trjExpensify** [comment]
Gotcha.

**Comment by trjExpensify** [comment]
>The title says Instant Submit, but from the OP & from a few comments, it doesn't look like the bug is related to instant submit?

I can't recall why we associated it with the `[Instant Submit]` project, but down to remove that association.

 >The OP says "Wait until the report is auto-submitted (i.e end of the day if set to Daily)" - this refers to Harvesting, right? I don't know if we have any truly automatic submitting other than "Instant submit" or harvesting, do we? Could be nice to clear that up?

Yeah, I believe the process of an `open` report being automatically submitted on a particular cadence governed by the "Scheduled submit" featuee is still called "harvesting" in the code base. I don't know what you're suggesting to be updated, but go ahead and edit accordingly if necessary. 👍 

>Also if this is truly a HIGH priority should we try to get someone to jump on this ASAP since this has been open since Jan? 😅 I can jump in possibly tomorrow if needed!

Now that we're working on Control, this is more important to fix yeah. 👍

**Comment by Beamanator** [comment]
Noice, thanks! And ya actually "harvesting" vs "auto-submit" - both are probably good. I got slightly scared when I was thinking of auto-approve, but that's obviously not the same 😅

**Comment by trjExpensify** [comment]
Gotcha. Yeah, different feature that for sure!

**Comment by trjExpensify** [comment]
Did you end up looking at this, Alex?

**Comment by Beamanator** [comment]
Huh so tried testing on dev a bit today

> On auto-submission, the the submit button should be removed.

I see the submit button being removed from the report preview AND from the report itself in NewDot 🤔 

https://github.com/user-attachments/assets/e202c2d7-2f37-4a56-b79d-1c6e0fe4368e

> The NewDot reportAction for autoSubmission should be added to the chat from the submitter, the OldDot one from Concierge should not appear in NewDot.

You can also see in the screenshot above that the report action that shows up is `submitted EGP 9.00`, and from the user (not from concierge)
- **Though it does not have the "... Via ScheduledSubmit" ending, is that needed?**

> On auto-submission of the report, the workspace chat should be unread for the approver, the LHN chat row should have a GBR green dot and an approve button should be in the preview component.

I'm also seeing the GBR show up for the approver when the report is auto-submitted / harvested:


https://github.com/user-attachments/assets/9205f324-53d5-4ba7-9219-47614339d577

**Comment by Beamanator** [comment]
@trjExpensify can you confirm if these bugs really still exist / need to be fixed? 😅

**Comment by trjExpensify** [comment]
Ah, that's awesome!! 

>Though it does not have the "... Via ScheduledSubmit" ending, is that needed?

Yeah, so this seems like the only outstanding piece to align on in the clean-up. I think it's helpful to distinguish between an automated submission and a manual action. Looking at a few alts:

<img width="859" alt="image" src="https://github.com/user-attachments/assets/fb054a38-f2a0-4641-bee2-53012c2587a5">

I think I like the Expensify user the best and using the pattern being introduced with copilot actions. Thoughts, @Expensify/design @JmillsExpensify?

**Comment by dubielzyk-expensify** [comment]
I think I prefer the second option personally

**Comment by JmillsExpensify** [comment]
I think I also like the Expensify user best, since it's a configuration, though that said, I just realized that for auto-reimbursement, we currently use the left-most pattern, like so. 

![CleanShot 2024-07-24 at 22 33 33@2x](https://github.com/user-attachments/assets/454099d8-2190-48cb-9658-577342715d28)

**Comment by JmillsExpensify** [comment]
So as a result, I think there's a strong argument that any of the left ones are most consistent with what's currently in product. I agree with @dubielzyk-expensify that the second option is probably best of those.

**Comment by shawnborton** [comment]
Nice one! 

So in the case where an action happens, I think I like just leading off with a clean statement of what happened, and then using parenthesis to describe if anything magical caused it? Like:

> **Shawn Borton**
> submitted $15.00

> **Shawn Borton**
> paid $15.00

and then

> **Shawn Borton**
> submitted $15.00 (automatically via delayed submission)

> Shawn Borton
> paid $15.00 (automatically via workspace rules)

I also like the idea of trying to reuse the same words/feature name we have in workspace settings too if possible. So if we're calling it delayed submission in the workspace editor, I like reusing that here too. Thoughts?

**Comment by shawnborton** [comment]
True, I think you make some good points and I like what you laid out!

**Comment by JmillsExpensify** [comment]
I kind of prefer no parentheses. I think Danny makes a good point that most other messages read like a normal sentence. 

As for using the feature names. That probably makes sense, though I do have reservations because they can still be jargony despite our best intentions. I wonder if we should hyperlink the names and link to a help page?

**Comment by trjExpensify** [comment]
I think the hyperlinks to the help page would be helpful.

**Comment by dannymcclain** [comment]
I'm definitely not against adding some links if you all think they could be helpful.

**Comment by JmillsExpensify** [comment]
I think it'd make me more comfortable. Delayed submission is pretty jargony. Workspace less so, but even than and rules together doesn't necessarily make it clear what it is.

**Comment by dannymcclain** [comment]
So that may give us something like this then?

>**Shawn Borton**
automatically submitted $15.00 via [delayed submission](url)

>**Shawn Borton**
automatically paid $15.00 via [workspace rules](url)

**Comment by JmillsExpensify** [comment]
Looks great to me!

**Comment by Beamanator** [comment]
Cool cool thanks so much for everyone's thoughts so far!! Seems like the current leader is:

>**Shawn Borton**
automatically submitted $15.00 via [delayed submission](url)

>**Shawn Borton**
automatically paid $15.00 via [workspace rules](url)

Any last thoughts before we move forward with this, @trjExpensify @shawnborton ?

**Comment by shawnborton** [comment]
That works for me. Do we have help articles ready for those features yet?

**Comment by trjExpensify** [comment]
Works for me too! I doubt we have articles for them yet, only Classic articles:

https://help.expensify.com/articles/expensify-classic/reports/Automatically-submit-employee-reports
https://help.expensify.com/articles/expensify-classic/workspaces/Configure-Reimbursement-Settings

As we've now allowed `Delayed submission` on collect again, we could probably have that resource created now. CC: @strepanier03 for how we go about getting that assigned out.

For workspace rules, that's a bit tougher, as we haven't added this config to NewDot yet, so it has to be configured in OldDot. So we can only point to the Classic article for now.

**Comment by Beamanator** [comment]
Cool cool! Maybe we can link to the old ones for now, and update to the new articles once they're ready.

Also to be clear, do we want basically the same report action messages to appear in OldDot? Example:

NewDot:
> **Shawn Borton**
> automatically submitted $15.00 via [delayed submission](https://github.com/Expensify/App/issues/url)

OldDot:
> Shawn Borton: automatically submitted $15.00 via [delayed submission](https://github.com/Expensify/App/issues/url)

**Comment by Beamanator** [comment]
Ok I don't meeeean to keep delaying working on this, BUT it looks like @marcochavezf just submitted [this PR](https://github.com/Expensify/Web-Expensify/pull/42989) to fix [this issue](https://github.com/Expensify/Expensify/issues/415522) related to auto-approvals, where the outcome is:

> Verify the message `Concierge final approved this report on behalf of <approver>` is shown in OldDot and `approved $x.x in NewDot.`

But from our discussion above, what i would expect for "auto-approvals" is:

NewDot:
>**Shawn Borton**
automatically approved $15.00 via [workspace rules](url)

OldDot:
>**Shawn Borton**
Shawn Borton: automatically approved $15.00 via [workspace rules](url)

Thoughts? 😅

**Comment by marcochavezf** [comment]
> Ok I don't meeeean to keep delaying working on this, BUT it looks like @marcochavezf just submitted https://github.com/Expensify/Web-Expensify/pull/42989 to fix https://github.com/Expensify/Expensify/issues/415522 related to auto-approvals,

fwiw, we're displaying `approved $x.x` in both OldDot and NewDot, and this change is to show the old wording in OldDot since it's impacting customers for their internal audits

**Comment by Beamanator** [comment]
Thanks for more context @marcochavezf - I just wonder if we should consider all of these together instead of separately

**Comment by trjExpensify** [comment]
I'm personally fine with leaving the OldDot version untouched and just focusing on the NewDot behaviour which is: 

>**Shawn Borton**
>automatically approved $15.00 via [workspace rules](https://github.com/Expensify/App/issues/url)

**Comment by Beamanator** [comment]
Sounds fair to me 👍

**Comment by Beamanator** [comment]
Hasn't been super high priority yet AFAIK, hoping to get to this next week

**Comment by Beamanator** [comment]
I plan to start taking a crack at these this week

**Comment by Beamanator** [comment]
PR for "Automatically submitted `<amount>` via delayed submission" created & ready for review ^ 👍 

For auto-reimbursed, I see we use a report action `"REIMBURSED"` but that isn't used in NewDot yet 🤔  I see there's multiple ways we can create that action (the Reimburse flow in OldDot can lead to multiple variations of messages showing in OldDot).

Anyone know if we're supporting that report action elsewhere or if this is the first place we'll be supporting it? If this is the first place we're supporting it, I wonder if we should also consider how we show that report action in NewDot when manual reimbursement happens in OldDot 🤔

**Comment by trjExpensify** [comment]
>For auto-reimbursed, I see we use a report action "REIMBURSED"

Are you saying on OldDot there's no distinct report action for a report paid via automation? 🤔

**Comment by Beamanator** [comment]
> Are you saying on OldDot there's no distinct report action for a report paid via automation? 🤔

Bingo - same with auto-submitted - we reuse report actions that are for the same "action" but we just store specific data in the report actions indicating if the actions were taken manually or automatically - i can link some code if you're interested to see it 🙏

**Comment by Beamanator** [comment]
Ok so little status update on the auto-reimbursement side of things:
1. Auto reimbursing in NewDot currently goes through 3 different flows:
    1. Control policy on advanced approvals
        - For this, we had to update the auth token used for auto-reimbursing [here](https://github.com/Expensify/Auth/pull/12489)
    2. Policies on instant submit
    3. All others
2. For **control policy on advanced approvals**, we should be good to go to start modifying the report action 👍 
3. For **Policies on Instant Submit** we currently call `PayMoneyRequest` to do auto-payments
    - This command adds the NewDot-needed report action showing the report payments 👍 
4. For **"all other auto-submitting"** we currently call `ReimburseReport` to do the auto-payments
    - Sadly, this command does NOT add the NewDot-needed "IOU -> Pay" report action, instead we only add the `Reimbursed` report action which we plan NOT to support in NewDot - so I need to figure out how to get the "IOU -> Pay" report action added in that flow.

Another point: How do we want to update the Report Preview here, when the report was paid automatically?
- AFAIK this is what it currently looks like after a report is reimbursed - are we ok with keeping this or do we want to try to make it clear that the report was paid automatically?:

<img width="427" alt="Screenshot 2024-09-13 at 1 52 38 PM" src="https://github.com/user-attachments/assets/a07eb521-11e3-43ce-877d-0d0be57774f7">

**Comment by trjExpensify** [comment]
>AFAIK this is what it currently looks like after a report is reimbursed - are we ok with keeping this or do we want to try to make it clear that the report was paid automatically?:

No change. 👍

**Comment by Beamanator** [comment]
That makes my life easy <3

**Comment by Beamanator** [comment]
Web-E & Auth PRs for auto-reimbursed copy in review ^

**Comment by Beamanator** [comment]
Web-E & Auth PRs are merged 👍  Two App PRs left:
1. https://github.com/Expensify/App/pull/48939 - fixing some lint errors & waiting for C+ to finish testing (hard b/c it needs a report to be harvested)
2. https://github.com/Expensify/App/pull/49187 - getting close, but waiting for Web-E PR to get to prod

**Comment by Beamanator** [comment]
Ooh @trjExpensify - @mollfpr had a good question: Should we also update the "Submit & Close" report action to read like this?

> **Shawn Borton**
> automatically submitted and closed $15.00 via delayed submission

Also, I just realized I haven't touched "auto approval" report actions yet, that will come after the above action updates

**Comment by trjExpensify** [comment]
>Ooh @trjExpensify - @mollfpr had a good question: Should we also update the "Submit & Close" report action to read like this?

Hm, how does the OldDot one read? I'm not sure if we need to differentiate really, we don't call it "closed" in NewDot.

**Comment by Beamanator** [comment]
> Hm, how does the OldDot one read?

We currently just add the "(automatically closed due to submit-only policy)" part in OldDot signifying it's submit & closed

<img width="915" alt="Screenshot 2024-09-26 at 1 33 42 PM" src="https://github.com/user-attachments/assets/d1d148e1-2e68-45f5-8c1d-d6785887ce36">

**Comment by trjExpensify** [comment]
Yeah, interesting. We don't use "closed" terminology, so it would be strange to randomly chuck that in here as it's a foreign concept to a NewDot user. I vote for just using this one for now:

>automatically submitted $50.00 via delayed submission 

The nextSteps illustrates that it's finished and there's nothing more to be done.

**Comment by JmillsExpensify** [comment]
100% agree with Tom.

**Comment by Beamanator** [comment]
Giggidy 👍 I 100.5% agree with Tom 👍

**Comment by Beamanator** [comment]
Okkkkkkkk little update:
1. [Submitted via delayed submission](https://github.com/Expensify/App/pull/48939) should be in staging ✅ 
2. [Paid via workspace rules](https://github.com/Expensify/App/pull/49187) is very slow to get reviewed 🔴 
3. [Submitted and closed](https://github.com/Expensify/App/pull/49905) (showing same as "Submitted via delayed submission") has PRs in review 🟡 
    - 2nd PR in review: https://github.com/Expensify/Auth/pull/12645
4. Approved via workspace rules coming next 🙏

**Comment by Beamanator** [comment]
Okkk another question about auto-approval 😬 
- In OldDot, report action looks like: `Concierge final approved this report on behalf of zz@gmail.com`
- In NewDot, we'll make this read `automatically approved $1.00 via workspace rules` (detailed [here](https://github.com/Expensify/App/issues/35091#issuecomment-2270134054))
- In NewDot, the Report Preview component (in a workspace chat) currently looks like:
    - <img width="426" alt="Screenshot 2024-09-29 at 9 41 02 PM" src="https://github.com/user-attachments/assets/1debeb7c-07bc-4140-b4f0-98066ae13f04">
    - **Question:** Do we want this to remain the same? Or include `automatic` / `auto-approved` somehow?

**Comment by trjExpensify** [comment]
>Question: Do we want this to remain the same? Or include automatic / auto-approved somehow?

Nah, we don't want to include that in the report preview component. It's just

`workspaceName approved:`

**Comment by trjExpensify** [comment]
>Oh and Question 2: Do we need to "optimistically" handle auto-approval & payment in this issue or can we do that in a new one / later? :D

How would it work offline if we don't? 😕

**Comment by Beamanator** [comment]
Exactly :D but the scope of this issue has continued to grow AND offline should be able to be worked externally, so I'd prefer if we do that in a new issue 🤷

**Comment by Beamanator** [comment]
Question about auto-reimbursed report action here too :D https://github.com/Expensify/App/issues/50090#issuecomment-2389766012

**Comment by Beamanator** [comment]
There's a decent amount of PRs for this issue so that ^ checklist probs won't be the last 😅

**Comment by trjExpensify** [comment]
What's the dealio with this project and payments?

**Comment by Beamanator** [comment]
Pshhhh you don't wanna search through all 168 hidden messages to find linked PRs & who's owed what? 😅 

I'll help look tomorrow 😳 🙈

**Comment by trjExpensify** [comment]
Lol, I more so meant is there 1) some kind of price agreed with the C+ on this for the wider project or is it standard and 2) are regression tests being added centrally for this project.

**Comment by trjExpensify** [comment]
>I'll help look tomorrow 😳 🙈

I'll certainly take it though!

**Comment by Beamanator** [comment]
Shit well I didn't have time to help today, sry :/ but yeah good question - so far no wider price agreement elsewhere so I thinkkkk it should be paid standard per PR reviewed

**Comment by trjExpensify** [comment]
Great, so then of the gazillion and one PRs for this issue... narrowing it down to the App PRs only, are any of these additional App PRs considered regressions? 

- #48939
- #49187
- #49905
- #49909
- #50181

**Comment by Beamanator** [comment]
Personally I wouldn't consider any regressions, I just split up the work into many PRs 😅

**Comment by trjExpensify** [comment]
Okay, so confirming payments will be as follows:

- https://github.com/Expensify/App/pull/48939 - $250 to @mollfpr for the C+ review 
- https://github.com/Expensify/App/pull/49187 - no payment due, only Tyler reviewed. 
- https://github.com/Expensify/App/pull/49905 - $250 to @mollfpr for the C+ review 
- https://github.com/Expensify/App/pull/49909 - $250 to @mollfpr for the C+ review
- https://github.com/Expensify/App/pull/50181 - no payment due to Ishpaul, payment is being handled in https://github.com/Expensify/App/issues/50090

So in total that's $750 to @mollfpr for the C+ reviews here. 

With that cleared up, let's talk about the checklist. At the end of the day, what we ended up doing here was an improvement rather than a regression. That being to improve the auto-submit/approver/pay reportActions copy after introducing automation. So I don't think we have a RCA to chase. 

We should add/update regression tests for these cases, so @mollfpr can you suggest them for: 

- auto-submitted
- auto-approval
- auto-payment

Thanks!

**Comment by mollfpr** [comment]
**auto-submitted**

1. Create a policy with delayed submission (example: Daily) & simple or advanced approvals
2. Create a report & a few expenses on that policy
3. Let report auto-harvest
4. Verify the "Submitted" report action now reads automatically submitted <amount> via delayed submission where the "Delayed Submission" test links out to https://help.expensify.com/articles/expensify-classic/reports/Automatically-submit-employee-reports

**auto-approval**

1. Create a collect policy in NewDot
2. In OldDot, set up auto approvals like so:
    - In the workspace Members tab, set "Approval Mode" to "Submit & Approve"
    - In the Manually approve all expenses over the field, enter a number like $200.00 (something over $100)
    - In the Randomly route reports for manual approval: field, enter 0%
3. Set the policy's "scheduled submit" to "on" and to anything other than "Instant" (a.k.a. "Daily")
4. As the submitter, submit an expense for $101.00 (something between $100 and $200)
5. Verify the expense gets auto-approved and the text on LHN and report action is `automatically approved $<amount> via workspace rules`

**auto-approval and auto-forwarding**

1. Set up Control policy on NewDot, with Advanced Approvals enabled - non instant submit
2. Make sure the policy has at least 3 members, where at least one employee has an approval chain like this:
    - Employee A submitsTo Employee B
    - Employee B forwardsTo Employee C
3. In OldDot as the policy admin, enable auto approvals
    - In the workspace Members page, set Manually approve all expenses over: to something like $1000
    - Set Randomly route reports for manual approval: to 0%
4. In NewDot, have Employee A submit an expense in their workspace chat
5. Verify the expense is auto approved from Employee B
6. On Employee C, open the workspace chat for Employee A and verify an expense submitted from Employee A is forwarded to Employee C
7. Approved the expense
8. Verify the text on LHN and report action is `approved <amount>`

**auto-payment**

1. Create a Collect policy, on Instant submit, with Submit & Approve enabled
2. In OldDot, in the workspace Members tab, set Manually approve all expenses over: to $0.01
3. In OldDot, in the workspace Reimbursement tab, set reimbursement to Direct and check the Manual Reimbursement section - you should see that Reports above $100.00 are not reimbursed unless manually initiated.
4. Create a withdrawal-ready bank account for the workspace
5. Create a deposit-ready bank account for the workspace owner (personal bank account shouldn't be linked to the workspace
6. In OldDot, as the workspace owner, create a report & add an expense of $1, then approve the report
7. Now open that report in NewDot and verify you see the new "automatically paid" report action copy
8. Now in NewDot, go to the workspace, submit an expense, then approve that report
9. Verify that report also gets auto-reimbursed & shows the new report action copy

**Comment by garrettmknight** [comment]
$750 approved for @mollfpr based on [this summary](https://github.com/Expensify/App/issues/35091#issuecomment-2433584743)

**Comment by trjExpensify** [comment]
@mollfpr I'm not following this test "auto-approval and auto-forwarding"? If auto-approval is enabled in that case, why is the report action `approved <amount>`?

**Comment by mollfpr** [comment]
@trjExpensify I believe Employee C can't auto-approve the expense and the report action is expected to `approved <amount>`. Is that answer your question?

**Comment by trjExpensify** [comment]
Huh, why not? 🤔 If auto-approval is enabled and the report isn't flagged for review because of the amount and/or random audit percentage, it should be final approved.

**Comment by mollfpr** [comment]
Maybe @Beamanator can help to answer it 😅

 From your above comment, I have the same thought now.

**Comment by Beamanator** [comment]
Huh, Idk if I wrote that initially but I'd definitely expect that to say "`automatically approved ${amount} via <a href="${CONST.CONFIGURE_REIMBURSEMENT_SETTINGS_HELP_URL}">workspace rules</a>`,"

Code ref: https://github.com/Expensify/App/blob/7b9a0cd644ea36e1543ae63a23ea72f610029dc4/src/libs/OptionsListUtils.ts#L638

**Comment by trjExpensify** [comment]
Okay, @mollfpr can we modify the tests then? Do we scrap that second one completely or what else are we including it for?

**Comment by trjExpensify** [comment]
Sorry, I still don't get it. When auto-approval is enabled, the report is auto-approved at every step in the approval chain, not just the first one. So how is this happening and/or an accurate test?

> **auto-approval and auto-forwarding**
> 
> 1. Set up Control policy on NewDot, with Advanced Approvals enabled - non instant submit
> 2. Make sure the policy has at least 3 members, where at least one employee has an approval chain like this:
>     - Employee A submitsTo Employee B
>     - Employee B forwardsTo Employee C
> 3. In OldDot as the policy admin, enable auto approvals
>     - In the workspace Members page, set Manually approve all expenses over: to something like $1000
>     - Set Randomly route reports for manual approval: to 0%
> 4. In NewDot, have Employee A submit an expense in their workspace chat
> 5. Verify the expense is auto approved from Employee B and the text on LHN and report actions is `automatically approved $<amount> via workspace rules`
> 6. On Employee C, open the workspace chat for Employee A and verify an expense submitted from Employee A is forwarded to Employee C
> 7. Approved the expense
> 8. Verify the text on LHN and report action is `approved <amount>`

**Comment by Beamanator** [comment]
oh uhhhhh we might not have built multi-auto-approvals yet 🙃

**Comment by trjExpensify** [comment]
>oh uhhhhh we might not have built multi-auto-approvals yet 🙃 https://github.com/Expensify/Expensify/issues/434467

Wowza, I did not know that. So then I don't think we add a test for the temporary behaviour. Rather, when we add support for auto-approval with multi-level approvals, add a test then to verify both levels are auto-approved. 

I've gone ahead and created a GH for 1, 2 and 4 above for now. 

Go ahead and request $750 on the basis of [this payment summary](https://github.com/Expensify/App/issues/35091#issuecomment-2433584743), @mollfpr. Closing!
